src/commands/new.rs

```diff
@@ -1,36 +1,98 @@
 use cargo_metadata::camino::Utf8PathBuf;
+#[cfg(feature = "fetch-template")]
 use directories::ProjectDirs;
-use log::{debug, error, info};
+use log::{debug, info, warn};
+use serde_json::Value;
 
 use crate::errors::CliError;
 use std::{
     fs, io,
     path::{Path, PathBuf},
 };
 
+#[derive(Debug, Clone)]
+struct Template {
+    pub data: Vec<u8>,
+    pub sha: Option<String>,
+}
+
 #[cfg(feature = "fetch-template")]
-async fn fetch_template() -> reqwest::Result<Vec<u8>> {
-    info!("Fetching template...");
+async fn get_current_sha() -> Result<String, CliError> {
+    let client = reqwest::Client::new();
+    let response = client
+        .get("https://api.github.com/repos/vexide/vexide-template/commits/main?per-page=1")
+        .header("User-Agent", "vexide/cargo-v5")
+        .send()
+        .await;
+    let response = match response {
+        Ok(response) => response,
+        Err(err) => return Err(CliError::ReqwestError(err)),
+    };
+    let response_text = response.text().await.ok().unwrap_or("{}".to_string());
+    match &serde_json::from_str::<Value>(&response_text).unwrap_or_default()["sha"] {
+        Value::String(str) => Ok(str.clone()),
+        _ => unreachable!("Internal error: GitHub API broken"),
```

Comment:
I don't think this should be `unreachable` or a panic - as unlikely as it is I don't entirely trust serde/Github to be totally reliable here.

Reply:
Agreed. Unreachable is completely incorrect here anyway.